libmscore/style.cpp

```diff
@@ -2474,18 +2526,217 @@ void MStyle::save(XmlWriter& xml, bool optimize)
             _chordList.write(xml);
             xml.etag();
             }
-      xml.tag("Spatium", value(Sid::spatium).toDouble() / DPMM);
+      xml.tag("Spatium", value(Sid::spatium).toDouble() / DPMM);                 // 3.01
+      xml.tag("spatium", rint(value(Sid::spatium).toDouble() / DPI_F * MSCX_F)); // 3.01+ last one wins at load time
       xml.etag();
       }
 
+//--------------------------------------------------------------------------
+//   initPageLayout - only applied to MScore::_baseStyle and _defaultStyle
+//                    must wait until setMscoreLocale() is called in main()
+//--------------------------------------------------------------------------
+
+void MStyle::initPageLayout()
+      { // by default, units are millimeters or inches, depending on locale
+        // default preferences are initialized prior to default styles in main()
+      QPageLayout::Unit     unit;
+      QPageSize::PageSizeId psid;
+      if (MScore::testMode) {
+            unit = QPageLayout::Millimeter;
+            psid = QPageSize::A4;
+            }
+      else {
+            unit = QPageLayout::Unit(MScore::unitsValue());
+            psid = MStyle::isMetric(unit) ? QPageSize::A4
+                                          : QPageSize::Letter;
+            }
+      _pageSize = QPageSize(psid);           
+      _pageOdd  = MPageLayout();
```

Comment:
Please, don't use extra spaces to visually align the variables. It will be a pain in ass for anyone who uses auto-formatter.

Reply:
@sidewayss please address the comment.